cmd/auth-provider-gcp/provider/provider.go

```diff
@@ -43,7 +43,8 @@ const (
 func MakeRegistryProvider(transport *http.Transport) *gcpcredential.ContainerRegistryProvider {
 	httpClient := makeHTTPClient(transport)
 	provider := &gcpcredential.ContainerRegistryProvider{
-		MetadataProvider: gcpcredential.MetadataProvider{Client: httpClient},
+		MetadataProvider:                  gcpcredential.MetadataProvider{Client: httpClient},
+		SkipContainerRegistryUrlsMatching: true,
```

Comment:
```suggestion
		UseRegistryFromImage: true,
```

Reply:
Updated both files by using the suggestion. Thanks!